### DISCUSSION TITLE
Internal package support?

### DISCUSSION THREAD

**Discussion by jbreitbart** [tectonic-typesetting/tectonic#1001] [Q&A]
Hi

I am wondering if there is a way to use non-public packages. I have a company internal package stored that is provided as a "TDS compliant archive". Is there a way to use that together with tectonic? Sorry, I am really bad with the internals of LaTeX :disappointed: I tried to copy the files of the archive in the same folder as my tex file, but this gets very messy and I got other error messages as well.

**Comment by MTRNord** [comment]
I was about to ask the same :)

Basically we are currently porting various old PowerPoint templates to beamer, so I wonder what the best way is to package them and then use them in tectonic. They for obvious reasons wouldn't be on CTAN. So I wonder what the correct way of handling this would be.

**Comment by pkgw** [comment]
If you're working above a certain scale, I'd suggest creating your own bundle that includes the files. The [tectonic-texlive-bundles][1] repo provides a reproducible workflow for doing so, and it shouldn't be hard to customize. The only really hassle there is getting a checkout of the many-gigabyte Git mirror of the TeX Live SVN repo. You'd have to put the resulting several-gigabyte bundle file online somewhere, but that's not asking too much these days.

[1]: https://github.com/tectonic-typesetting/tectonic-texlive-bundles/

If you're not able to go that route, the solution isn't as clean as I'd like, since the right approach is basically to integrate a package manager into Tectonic, which is pretty big engineering lift. You can symlink the relevant files into your projects, or just copy them in every time, or use the [`-Z search-path`][2] option for one-off builds ("V1" CLI or `tectonic -X compile`).

[2]: https://tectonic-typesetting.github.io/book/latest/v2cli/compile.html#unstable-options

But if anyone wants extra credit for that engineering lift: teach Tectonic to recognize NPM `package.json` files and/or Conda `environment.yml` exports and search the associated package trees for additional support files. Integration with the latter would also give us a path for solving the longstanding biblatex/biber integration issues.
